nixos/modules/services/monitoring/fluent-bit.nix

```diff
@@ -0,0 +1,107 @@
+{
+  config,
+  lib,
+  pkgs,
+  ...
+}:
+let
+  cfg = config.services.fluent-bit;
+
+  yamlFormat = pkgs.formats.yaml { };
+
+  configurationFile =
+    if (cfg.configurationFile == null) then
+      (yamlFormat.generate "fluent-bit.yaml" cfg.configuration)
+    else
+      cfg.configurationFile;
+in
+{
+  options.services.fluent-bit = {
+    enable = lib.mkEnableOption "Fluent Bit";
+    package = lib.mkPackageOption pkgs "fluent-bit" { };
+    configurationFile = lib.mkOption {
+      type = lib.types.nullOr lib.types.path;
+      default = null;
+      description = ''
+        Fluent Bit configuration. See
+        <https://docs.fluentbit.io/manual/administration/configuring-fluent-bit/yaml>
+        for supported values.
+
+        {option}`configurationFile` takes precedence over {option}`configuration`.
+
+        Note: Restricted evaluation blocks access to paths outside the Nix store.
+        This means detecting content changes for mutable paths (i.e. not input or content-addressed) can't be done.
+        As a result, `nixos-rebuild` won't reload/restart the systemd unit when mutable path contents change.
+        `systemctl restart fluent-bit.service` must be used instead.
+      '';
+      example = /etc/fluent-bit/fluent-bit.yaml;
+    };
+    configuration = lib.mkOption {
+      type = yamlFormat.type;
+      default = { };
+      description = ''
+        See {option}`configurationFile`.
+
+        {option}`configurationFile` takes precedence over {option}`configuration`.
+      '';
+      example = {
+        service = {
+          grace = 30;
+        };
+        pipeline = {
+          inputs = [
+            {
+              name = "systemd";
+              systemd_filter = "_SYSTEMD_UNIT=fluent-bit.service";
+            }
+          ];
+          outputs = [
+            {
+              name = "file";
+              path = "/var/log/fluent-bit";
+              file = "fluent-bit.out";
+            }
+          ];
+        };
+      };
+    };
+    user = lib.mkOption {
+      type = lib.types.str;
+      default = "root";
+      description = ''
+        The user that runs Fluent Bit.
+      '';
+      example = "fluent-bit";
+    };
+    # See https://docs.fluentbit.io/manual/administration/configuring-fluent-bit/yaml/service-section.
+    grace = lib.mkOption {
+      type = lib.types.ints.positive;
+      default = 5;
+      description = ''
+        The grace time in `seconds` as an integer value. The engine loop uses a grace timeout to define the wait time before exiting.
+
+        The `service.grace` option in the Fluent Bit configuration should be ≤ this option.
+      '';
+      example = 30;
+    };
```

Comment:
This will only set `TimeoutStopSec` in the systemd service but not `service.grace` in the fluent-bit configuration file. While this is documented to a degree, the description does not mention this changes the timeout in the systemd service and thus kills the process. Calling the option the same as in the configuration makes this behavior pretty unintuitive.

IMHO this won't be a problem in most cases anyway since the default for `TimeoutStopSec` (or `DefaultTimeoutStopSec`) is 90s. If users change this, they have to deal with the consequences.

I would either remove this or explicitly mention in the documentation what it does and rename it accordingly.

Reply:
Thanks, that's better. :+1: